frontend/src/layout/AppLayout.vue

```diff
@@ -93,13 +94,25 @@ const handleUpdateMinCPU = (val: any) => {
 
 const router = useRouter();
 watch(
-  () => [searchConfigStore.searchConfig],
+  () => searchConfigStore.searchConfig,
   () => {
     const config = searchConfigStore.searchConfig;
-    router.replace({
+    const queryParam: RouteParam = {
+      cloudProvider: config.cloudProvider,
+      region: config.region.join("-"),
+      engineType: config.engineType.join("-"),
+      chargeType: config.chargeType.join("-"),
```

Comment:
This brings a limitation that `region`, `engineType`, `chargeType` cannot contain the character `"-"`, please watch out.

Reply:
Regions can contain dash, e.g. us-central1 https://cloud.google.com/compute/docs/regions-zones